manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -1403,6 +1403,15 @@ rules:
   - apiGroups: ["authorization.k8s.io"]
     resources: ["subjectaccessreviews"]
     verbs: ["create"]
+  - apiGroups: [""]
```

Comment:
is there any way to turn this off? This effectively gives root admin access to the cluster - a scary proposition when the role is called "reader". 

These permissions are only needed when following worst-practices - real users should never need these (public certs).

Reply:
TBH, I haven't really looked at what permissions are really necessary, maybe these aren't all needed. I think what is needed is enough permission for any istiod to aquire the leader lock on another cluster.

Reply:
We need to make sure users understand how to do this, in relation to the istiodless remote cluster install docs. This means linking docs for setting up a proper CA (or file mounted certs) on the istiod side, and if they're not public certs tell them they somehow need to "install the root of trust on their system" on the remote cluster side. 

For testing, I think we can get a step closer to this. In the integ-suite-kind/bash layer we could generate certs and install them on the KinD nodes. We then pass flags to the Go part of the framework pointing to the certs. The istio-install logic can pick up on this to mount them and tell istiod to use them, similar to these tests: https://github.com/istio/istio/blob/f85509e8a90f4e7970524728c2fc99ebda7f11ed/tests/integration/security/file_mounted_certs/main_test.go#L140

Reply:
I think mounting into kind node is not sufficient as the pods also need the cert for XDS and they do not inherit it from the node?

Reply:
@stevenctl @howardjohn I turned these extra permissions back off by default. I also realized that they aren't really needed for enabling istiodless remotes anyway, because they are just allowing external istiods to control other primary clusters. We don't want external istiods to control other primaries in the primary-primary MC setup. In fact, that's why I fixed the leadership election to make sure the local istiod wins the election. Not having these permissions short circuits that election process, so it's fine. We only need to turn on these extra permissions if we want HA control across primary clusters, but that's out of scope for this PR, which is just trying to get istiod out of remote clusters in the currently documented MC setups.

Reply:
My point is users probably won't do the right thing unless we document it. If we document a version where we give these broad permissions to patch the ca bundle into remote clusters, people will just do that. 

If we tell users "do things properly" without steps, they'll report bugs here saying primary-remote is broken. 

But, whatever we document we should try to replicate in our testing.

Reply:
I see, you're concerned about doing this in remotes too. The current pattern is now the same for all remote clusters, whether controlled by a primary istiod or an external one. If you feel that there's a better approach that we should document (for both I assume), we do have 3 months to figure it out because this is not going into 1.14 and will only be in the preliminary docs until 1.15. I'm happy to help work on another ("do things properly") pattern, but will definitely need help to understand what that should be.